splice_si_api/src/main/java/com/splicemachine/si/impl/store/ActiveTxnCacheSupplier.java

```diff
@@ -33,16 +33,15 @@
  * Date: 6/18/14
  */
 public class ActiveTxnCacheSupplier implements TxnSupplier {
-    private final LongKeyedCache<TxnView> cache;
+    private final Cache<Long,TxnView> cache;
     private final TxnSupplier delegate;
 
-    public ActiveTxnCacheSupplier(TxnSupplier delegate, int maxSize) {
-        this(delegate, maxSize, false);
+    public ActiveTxnCacheSupplier(TxnSupplier delegate, int initialSize, int maxSize) {
+        this(delegate, initialSize, maxSize, false);
     }
 
-    public ActiveTxnCacheSupplier(TxnSupplier delegate, int maxSize, boolean threadSafe) {
-        this.cache = LongKeyedCache.<TxnView>newBuilder().maxEntries(maxSize)
-                .withHashFunction(HashFunctions.murmur3(0)).threadSafe(threadSafe).build();
+    public ActiveTxnCacheSupplier(TxnSupplier delegate, int initialSize, int maxSize, boolean threadSafe) {
+        this.cache = CacheBuilder.newBuilder().initialCapacity(initialSize).maximumSize(maxSize).concurrencyLevel(threadSafe ? 32 : 1).build();
```

Comment:
Where does 32 come from?

Reply:
It was arbitrary, I substituted it for Guava's default of 4 with an explicitly defined constant.